crates/data-plane-controller/tests/test_preview.rs

```diff
@@ -0,0 +1,71 @@
+use data_plane_controller::{controller, stack};
+use std::{
+    collections::{HashMap, VecDeque},
+    sync::{Arc, Mutex},
+};
+
+mod util;
+use util::{initial_state, mock_emit_log_fn, mock_run_cmd_fn, TraceEntry};
+
+#[tokio::test]
+async fn test() {
+    let trace = Arc::new(Mutex::new(Vec::new()));
+
+    let controller = data_plane_controller::Controller {
+        dns_ttl: std::time::Duration::ZERO,
+        dry_dock_remote: "git@github.com:estuary/est-dry-dock.git".to_string(),
+        ops_remote: "git@github.com:estuary/ops.git".to_string(),
+        secrets_provider: "testing".to_string(),
+        state_backend: "file:///tmp/pulumi-test-state".parse().unwrap(),
+        emit_log_fn: mock_emit_log_fn(trace.clone()),
+        run_cmd_fn: mock_run_cmd_fn(trace.clone()),
+    };
+
+    let mut state: Option<stack::State> = None;
+    let mut inbox: VecDeque<(models::Id, Option<controller::Message>)> = VecDeque::new();
+    let mut checkouts: HashMap<String, tempfile::TempDir> = HashMap::new();
+    let mut row_state = initial_state();
+
+    inbox.push_back((
+        models::Id::zero(),
+        Some(controller::Message::Start(row_state.data_plane_id)),
+    ));
+    inbox.push_back((
+        models::Id::zero(),
+        Some(controller::Message::Preview(controller::Preview {
+            branch: "preview-branch".to_string(),
+        })),
+    ));
+    inbox.push_back((models::Id::zero(), Some(controller::Message::Enable)));
```

Comment:
you can remove Enable and eliminate a lot of snapshot churn

Reply:
@jgraettinger I intentionally kept Enable just as an extra caution that `preview_branch` is not going to affect the branch used by other parts of the code, since that is a compliance concern